optax/transforms/_accumulation.py

```diff
@@ -334,7 +334,9 @@ def update(self,
     # Note: we do not enclose variables to allow JAX to re-use memory buffers.
     def _do_update(updates, state, params):
       acc_grads = jtu.tree_map(
-          lambda upd, acc: self._acc_update(upd, acc, n_acc=state.mini_step),
+          lambda upd, acc: self._acc_update(
```

Comment:
Now that I look at it again, a slightly better version would be the following as we can expect the leaves to be `jnp.array`s:
```python
acc_grads = jtu.tree_map(
          lambda upd, acc: self._acc_update(upd.astype(acc.dtype), acc, n_acc=state.mini_step)
          updates,
          state.acc_grads,
      )
```
and the following for line 365 ff.:
```python
final_updates = jtu.tree_map(
           lambda ga, st: emit * ga.astype(st.dtype),
           final_updates,
           state.acc_grads
       )
```
Let me know what you think of the proposed fix in general and I'll updates these.